Src/FluentAssertions.Net40/Xml/XmlReaderValidator.cs

```diff
@@ -0,0 +1,239 @@
+﻿using FluentAssertions.Execution;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml;
+
+namespace FluentAssertions.Xml
+{
+    class XmlReaderValidator
+    {
+        AssertionScope assertion;
+        XmlReader subjectReader, expectedReader;
+
+        string CurrentLocation
+        {
+            get { return "/" + string.Join("/", locationStack.Reverse()); }
+        }
+
+        Stack<string> locationStack = new Stack<string>();
+
+        public XmlReaderValidator(XmlReader subjectReader, XmlReader expectedReader, string because, object[] reasonArgs)
+        {
+            assertion = Execute.Assertion.BecauseOf(because, reasonArgs);
+        
+            this.subjectReader = subjectReader;
+            this.expectedReader = expectedReader;
+        }
+                
+        class ValidationResult
```

Comment:
Instead of collecting these validation errors yourself, you can also use an `AssertionScope` like [this](https://github.com/dennisdoomen/FluentAssertions/blob/master/Tests/FluentAssertions.Shared.Specs/AssertionScopeSpecs.cs#L54) to have `Execute.Assertion` add all assertion failures together.